src/esthri/src/lib.rs

```diff
@@ -549,41 +519,48 @@ where
     Ok(keys)
 }
 
-pub fn list_objects_stream<'a, T>(
+pub fn list_objects_stream<'a, T, SR0, SR1>(
     s3: &'a T,
-    bucket: &'a str,
-    key: &'a str,
-) -> impl TryStream<Ok = Vec<S3ListingItem>, Error = eyre::Error> + Unpin + 'a
+    bucket: SR0,
+    key: SR1,
+) -> impl TryStream<Ok = Vec<S3ListingItem>, Error = Error> + Unpin + 'a
 where
     T: S3 + Send,
+    SR0: AsRef<str> + 'a,
+    SR1: AsRef<str> + 'a,
 {
-    list_objects_stream_with_delim(s3, bucket, key, None)
+    list_objects_stream_with_delim(s3, bucket, key, None as Option<&str>)
```

Comment:
Maybe you could create a variable with a name representative of the parameter here.

Reply:
Done